src/dbnode/storage/bootstrap/bootstrapper/fs/migrator/options.go

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package migrator
+
+import (
+	"errors"
+
+	"github.com/m3db/m3/src/dbnode/namespace"
+	"github.com/m3db/m3/src/dbnode/persist/fs"
+	"github.com/m3db/m3/src/dbnode/persist/fs/migration"
+	"github.com/m3db/m3/src/dbnode/storage"
+	"github.com/m3db/m3/src/x/instrument"
+)
+
+var (
+	errMigrationTaskFnNotSet      = errors.New("migrationTaskFn not set")
+	errInfoFilesByNamespaceNotSet = errors.New("infoFilesByNamespaces not set")
+	errMigrationOptsNotSet        = errors.New("migrationOpts not set")
+	errInstrumentOptsNotSet       = errors.New("instrumentOpts not set")
+	errStorageOptsNotSet          = errors.New("storageOpts not set")
+	errFilesystemOptsNotSet       = errors.New("filesystemOpts not set")
+)
+
+type options struct {
+	migrationTaskFn      MigrationTaskFn
+	infoFilesByNamespace map[namespace.Metadata]fs.InfoFileResultsPerShard
+	migrationOpts        migration.Options
+	fsOpts               fs.Options
+	instrumentOpts       instrument.Options
+	storageOpts          storage.Options
+}
+
+// NewOptions return new migration opts
+func NewOptions() Options {
+	return &options{
+		migrationOpts:  migration.NewOptions(),
+		fsOpts:         fs.NewOptions(),
+		instrumentOpts: instrument.NewOptions(),
+		storageOpts:    storage.NewOptions(),
```

Comment:
nit: be careful with `NewOptions`, as some of these will create pools, e.g. https://github.com/m3db/m3/blob/master/src/dbnode/persist/fs/options.go#L118

Most of these aren't really big enough to worry about, but since you have Validate checks to ensure this is set, may be ok to not use them here?